### ISSUE TITLE
Can I intergate with software like Autocad or Revit ?

### ISSUE THREAD

**Issue by chuongmep** [dotnet/interactive#3167]
**The package and version I'm asking about**: 

### Question
 Hi team, I have an idea about Jupyter Notebook can open in Autodesk Revit or Autodesk Autocad and use language C# or Python to code, whether your project is good point to research about that ? 
Regard !

**Comment by colombod** [comment]
@brettfo what do you think?

**Comment by JDiLenarda** [comment]
I have worked the idea of embedding an interactive Kernel in an AutoCAD MAP plugin as it is done in the Connect-WPF example. Unfortunately, AutoCAD .NET API are still in .NET Framework 4.*, so it's a no go (as far as I know).

I upvoted #3144 with the hope it could allow that.

(for now, I use a plugin embedding an old version of F# Interactive. For people wondering what's the stake is, it's great to be able to test things without stopping and restarting AutoCAD each time you make a significant modification of code.)

**Comment by JDiLenarda** [comment]
@chuongmep 
I didn't know AutoCAD had finally moved on, sad to be stuck with 2021 version at work.
Connect-WPF (example of how to embed a kernel in an application) is in the sample directory of this repo.
Writing .NET AutoCAD plugins is documented in Autodesk community sites.

**Comment by chuongmep** [comment]
@JDiLenarda , yeah, they moved , you can quick look to this post : https://adndevblog.typepad.com/autocad/2023/08/call-for-action-next-release-of-autocad.html
I will try with your sample fist, I hope this issue still open to discuss more, because I want see this can help alot aec move a head

**Comment by chuongmep** [comment]
@JDiLenarda I can't try with #!dispatcher , it is problem ? 
![image](https://github.com/dotnet/interactive/assets/31106432/66311e55-9a54-437b-8f89-c44cf12a18af)

**Comment by JDiLenarda** [comment]
@chuongmep 
I'm afraid I can't help on this. See #2369.

**Comment by chuongmep** [comment]
Thank you @JDiLenarda , I think it is not problem not important with me now, I try go around first.😊

**Comment by chuongmep** [comment]
@JDiLenarda  it can work with .netstandard 2.0 as well ?

**Comment by JDiLenarda** [comment]
@chuongmep 
I just asked my company to get me an AutoCAD 2024 developper licence, hope I can work on this soon.

**Comment by chuongmep** [comment]
> @chuongmep I just asked my company to get me an AutoCAD 2024 developper licence, hope I can work on this soon.

2024 still is .net 4.8 @JDiLenarda , We just can try with beta version now and we also have beta license as well

**Comment by colombod** [comment]
> Thank you @JDiLenarda , I think it is not problem not important with me now, I try go around first.😊

There have been bug fixes in the WPF sample area, not sure if they help (we did a release yesterday). A lot of .net interactive libraries are netstandard 2.1. let us know how is it going with 4.8 interoperability. @brettfo is a CAD lover too and I would love to understand better you scenario. What are you achieving with .NET interactive interoperability with the cad app?

**Comment by colombod** [comment]
https://www.keanw.com/2023/08/the-next-release-of-autocad-and-net.html according to this you should be in a much better place with NET 6 and 8 support!

**Comment by chuongmep** [comment]
@colombod I checked with latest version beta and it working with .net6  as well , so many change about folder and also technology behind that. I'm trying hacking some thing now and will respone for everyone soon😂

**Comment by chuongmep** [comment]
Hi @JDiLenarda , I'm opened a project at here , hope we can try explore 
https://github.com/chuongmep/JupyterBIM
@colombod , What are you achieving with .NET interactive interoperability with the cad app? It will open the gate for Data enginner easier work with data from Autodesk software. With developer it really is gold because they can combine to explore programing depend on IDE visual studio code and jupyter notebook, just need write some line and share for another on the github with format ipynb as a guideline document.

**Comment by JDiLenarda** [comment]
@colombod 
I occasionally write AutoCAD extensions. AutoCAD.Net APIs are poorly documented, so it's a lot of trial and error to achieve what I want. Doing test with a project demands to close and restart AutoCAD a lot. Having an extension with an Interactive Kernel would help me to do quick tests from a Notebooks cell.

@chuongmep 
I will surely follow your project, but for now, there is a clause from the beta feedback program that me and my company deem too intrusive to subscribe.

**Comment by chuongmep** [comment]
Hi @colombod @JDiLenarda , any new update about downgrade version of dotnet to expand limit for this ?

**Comment by jonsequitur** [comment]
The core .NET Interactive package (which includes `ProxyKernel` infrastructure) now targets Netstandard 2.0.

**Comment by chuongmep** [comment]
Thank you for everyone helped , it can going now !
